public/app/plugins/datasource/prometheus/variables.ts

```diff
@@ -0,0 +1,55 @@
+import { from, Observable, of } from 'rxjs';
+import { map } from 'rxjs/operators';
+import {
+  DataQueryRequest,
+  DataQueryResponse,
+  rangeUtil,
+  StandardVariableQuery,
+  StandardVariableSupport,
+} from '@grafana/data';
+import { getTemplateSrv, TemplateSrv } from '@grafana/runtime';
+
+import { PrometheusDatasource } from './datasource';
+import { PromQuery } from './types';
+import PrometheusMetricFindQuery from './metric_find_query';
+import { getTimeSrv, TimeSrv } from '../../../features/dashboard/services/TimeSrv';
+
+export class PrometheusVariableSupport implements StandardVariableSupport<PrometheusDatasource> {
```

Comment:
Is there some comparison what it looked like before? Or didn't Prometheus support variables query before?

Reply:
This is brand new, Prometheus used metricFindQuery for the variable queries before. [With this change we still use metricFindQuery for the experimental tags query see my comment](https://github.com/grafana/grafana/pull/28217#issuecomment-714232445).